Marlin/src/gcode/calibrate/G33.cpp

```diff
@@ -508,6 +508,13 @@ void GcodeSuite::G33() {
 
   print_G33_settings(_endstop_results, _angle_results);
 
+  #if HAS_BED_PROBE
+    if (verbose_level != 0) {
+      delta_height += G33_offset - zprobe_zoffset;
+      G33_offset = zprobe_zoffset;
+    }
+  #endif
```

Comment:
This seems like it might cause some confusion if users are not specifically setting out to calibrate the `delta_height`. As already pointed out, the `zprobe_zoffset` should not be retroactively applied to previously-acquired values, because such attempts at cleverness tend to introduce unexpected side-effects.

For example, I note that in `settings.reset()` the `G33_offset` is set to the default `Z_PROBE_OFFSET_FROM_EXTRUDER`. However, on `settings.load()` it simply loads both the last-saved `G33_offset`, `delta_height`, and `zprobe_zoffset` values. These values will remain out-of-sync in perpetuity and will only be re-corrected here (the next time someone uses `G33`) with no guarantee that `M500` will follow. Generally speaking, it's not great to leave loose ends like this.

Reply:
"This seems like it might cause some confusion if users are not specifically setting out to calibrate the delta_height"
They should not be using a routine which sole purpose is to calibrate the delta_height etc... Are you serious?

G33_offset is specific to G33 and saves zprobe_zoffset (Z=0 position probing reported) the last time G33 was used, so next time G33 is used it can continue from that point. 

The default Z_PROBE_OFFSET_FROM_EXTRUDER corresponds to the default DELTA_HEIGHT etc. so G33 can start from that point.

"These values will remain out-of-sync in perpetuity"
That is indeed the problem caused by M851, until G33 is re-invoked: G33 brings them back in-sync. Since M851 no longer does the necessary postprocessing to keep values in-sync; G33 has to do it.

"tying delta_height to zprobe_zoffset is misleading"

There is physical relationship : 
- delta_height = (trigger point at homing - position of the nozzle at Z=0)
- zprobe_zoffset = (position of the nozzle at Z=0 - trigger point of the Z-probe)

- (trigger point at homing - trigger point of the Z-probe is constant)

The trigger points do not change, so if you change Z=0 by changing zprobe_zoffset and you don't change delta_height accordingly; you have 2 different points Z=0: one reported by probing depending on zprobe_zoffset and one used for printing depending on delta_height. G33 needs to get those Z=0 point back in-sync before it can do it's job.

Sorry if you find the physical relationships in Delta printers confusing and misleading.

But if the user wants to set these independently, no problem use M851, M665, M666,... but stay away from G33 since that routines CALIBRATES and sets all delta values as they should be, taking into account all relationships that exist between them.

"such attempts at cleverness tend to introduce unexpected side-effects"
Indeed a lot of cleverness is required to do proper delta calibration in a FW that was written for Cartesians and were Delta's were given the stepmother treatment. But in this case the unexpected side-effect of having 2 different Z=0 points was caused by M851.